libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/auth/RustMatrixAuthenticationService.kt

```diff
@@ -74,13 +79,19 @@ class RustMatrixAuthenticationService @Inject constructor(
             }
     }
 
-    override fun getHomeserver(): String? = authService.homeserverDetails()?.url()
+    override fun getHomeserver(): String = currentHomeserver
 
-    override fun getHomeserverOrDefault(): String = getHomeserver() ?: "matrix.org"
+    override fun getHomeserverDisplayValue(): String {
```

Comment:
DisplayValue in matrix module looks a bit weird, I'd keep this logic somewhere else (maybe presenter?)

Reply:
Wouldn't the presenter be re-created if we leave this screen? We need something that keeps this value for as long as the app is running. Would `rememberSaveable` work if the instantiated presenter is a new one?

Reply:
Well, actually, we need it for as long as the onboarding flow is displayed. Maybe we can create a SharedLoginPresenter that is passed from `LoginFlowNode` to its children?

Reply:
And to answer your question, no, rememberSaveable will only work if the Composable is restored, not recreated from navigation!